tests/analysis/test_analysis.py

```diff
@@ -6,21 +8,23 @@
 class TestAnalysisService(TestCase):
     def test_transform_for_projects_heatmap(self):
         data = [MockPipelineRun("run tests", "TestProject", "something"),
-                MockPipelineRun("run tests", "TestProject", "something"),
-                MockPipelineRun("re-verify env", "TestProject", "something-else"),
+                MockPipelineRun("re-verify env", "AnotherProject", "something-else"),
                 MockPipelineRun("re-verify env", "TestProject", "something-else"),
-                MockPipelineRun("re-verify env", "AnotherProject", "something-else")]
-        expected_heatmap_data = {"z": [[2, 2], [0, 1]],
+                MockPipelineRun("re-verify env", "AnotherProject", "something-else"),
+                MockPipelineRun("run tests", "TestProject", "something")]
+        expected_heatmap_data = {"z": [[2, 1], [0, 2]],
                                  "y": ["TestProject", "AnotherProject"],
                                  "x": ["run tests", "re-verify env"]}
+        shuffle(data)
```

Comment:
Cheap `quickcheck` :stuck_out_tongue: will look into replacing this with `hypothesis` tests soon I'd say.